### ISSUE TITLE
Example of dynamic argument in custom-directive.md is incorrect

### ISSUE THREAD

**Issue by oohira** [vuejs/vuejs.org#2155]
In #2062, an example of dynamic argument for custom directive was added. However, the intension is not clear because 

1. if it means [dynamic arguments introduced in v2.6](https://vuejs.org/v2/guide/syntax.html#Dynamic-Arguments), the example should be `mydirective:[argument]=dataproperty` instead of `mydirective:argument=[dataproperty]`
2. if it means just passing the argument dynamically, the example should be `mydirective:argument=dataproperty` instead of `mydirective:argument=[dataproperty]`. But this is already described in the previous example `v-demo:foo.a.b="message"`

I think the following example may be close to the original intention.

* https://jsfiddle.net/y0ftjLxs/

cc: @bk-dev @sdras

**Comment by phanan** [comment]
Pinging @sdras

**Comment by bk-dev** [comment]
If I'm totally honest, I was confused as well. Mainly because in my code base the syntax was different than what was outlined by @sdras. An example usage in my code base is `<div v-my-directive:[dynamicArg]></div>`. That said, I have never had a need to use `=dataproperty` to create statement.

My original goal was to pass a data property to a directive and when that data property changed the directive update hook would get invoked so the directive could react to the change. Doing something like 2) from @oohira's post would not have worked (at least not in my testing).

**Comment by sdras** [comment]
Yep, this was my fault for not reading through more thoroughly. I rewrote this section with clearer examples (thank you for the fiddle @oohira), but also rewrote the explanation to be more conversational in our docs style, and explain the problem space/use case. Let me know if that clears things up for you. Thanks for the issue!
